src/js/media-controller.js

```diff
@@ -447,75 +427,51 @@ const setAttr = (child, attrName, attrValue) => {
   return child.setAttribute(attrName, attrValue);
 };
 
-const isMediaSlotElement = (el) => el?.slot === 'media';
-const isMediaSlotElementDescendant = (el) => !!el.closest('*[slot="media"]');
-const hasDescendants = (el) => {
-  return !!(el?.childNodes?.length || el?.shadowRoot?.childNodes?.length);
-};
+const isMediaSlotElementDescendant = (el) => !!el.closest?.('*[slot="media"]');
 
-const shouldGetMediaUIElementDescendants = (el) => {
-  return !isMediaSlotElement(el) && hasDescendants(el);
+const isUndefinedCustomElement = (el) => {
+  const name = el?.nodeName.toLowerCase();
+  return name.includes('-') && !window.customElements.get(name);
```

Comment:
We unfortunately can't rely on `customElements.get(name)` for readiness of the element. See this note in my themes PR...
https://github.com/muxinc/media-chrome/blob/5d9773ca957b65a33e5f8dbaa882332bf2895bf5/src/js/media-controller.js#L415
It feels like a major bug, and was not easy to figure out what the hell was going wrong in a previous iteration of this.

What that means is we just have to always use the `whenDefined` promise. But that's not the worst and will simplify the code a little.

Reply:
Great callout. Will refactor and put a very wordy code comment where we're implementing the logic.